src/pages/home/report/ReportActionItem.js

```diff
@@ -19,30 +24,152 @@ const propTypes = {
 
     // Should the comment have the appearance of being grouped with the previous comment?
     displayAsGroup: PropTypes.bool.isRequired,
+
+    // Function to scroll the ReportActionsView to this item
+    scrollToThis: PropTypes.func.isRequired,
+
+    ...windowDimensionsPropTypes,
 };
 
-const ReportActionItem = props => (
-    <Hoverable>
-        {hovered => (
-            <View>
-                <View style={getReportActionItemContainerStyles(hovered)}>
-                    {!props.displayAsGroup
-                        ? <ReportActionItemSingle action={props.action} />
-                        : <ReportActionItemGrouped action={props.action} />}
-                </View>
-                <View style={styles.miniReportActionContextMenuWrapperStyle}>
-                    <ReportActionContextMenu
-                        reportID={props.reportID}
-                        reportActionID={props.action.sequenceNumber}
-                        isVisible={hovered}
-                        isMini
-                    />
-                </View>
-            </View>
-        )}
-    </Hoverable>
-);
+class ReportActionItem extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isModalVisible: false,
+        };
+
+        // The X and Y position (relative to the screen) where the popover will display.
+        this.popoverAnchorX = null;
+        this.popoverAnchorY = null;
+
+        // The width and height of the ReportActionContextMenu popover.
+        this.popoverWidth = null;
+        this.popoverHeight = null;
+
+        this.showModal = this.showModal.bind(this);
+        this.hideModal = this.hideModal.bind(this);
+        this.measurePopover = this.measurePopover.bind(this);
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        // Use underscore's isEqual to do a deep comparison of props.
+        // We can't use PureComponent because its shallow comparison wouldn't detect changes in the `action` prop.
```

Comment:
I'm not sure I understand the question here, but before we had this:

```javascript
export default memo(ReportActionItem, (prevProps, nextProps) => (
    prevProps.displayAsGroup === nextProps.displayAsGroup && _.isEqual(prevProps.action, nextProps.action)
));
```

But that is more complicated than it needed to be, becuase `_.isEqual` does a deep comparison anyways, so it could have just been:

```javascript
export default memo(ReportAcitonItem, (prevProps, nextProps) => _.isEqual(prevProps, nextProps));
```

And then in our updated component we now have a state variable too, so our `shouldComponentUpdate` looks like this:

```javascript
shouldComponentUpdate(nextProps, nextState) {
    // Use underscore's isEqual to do a deep comparison of props.
    // We can't use PureComponent because its shallow comparison wouldn't detect changes in the `action` prop.
    return !_.isEqual(this.props, nextProps) || !_.isEqual(this.state, nextState);
}
```

Reply:
TBH I feel like I'm having deja vu and thought we already decide that what we had before worked fine and that this change isn't necessary.

>that is more complicated than it needed to be, becuase _.isEqual does a deep comparison anyways

Technically it evaluates faster and avoids doing a deep comparison so they are not 100% equivalent.

Reply:
Okay, well the change is necessary because this is now a class component and has state, but I can make it so that it doesn't always do a deep comparison.